test/built-ins/BigInt/asUintN/order-of-steps.js

```diff
@@ -0,0 +1,32 @@
+// Copyright (C) 2017 Josh Wolfe. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: pending
+description: BigInt.asUintN order of parameter type coercion
+info: >
+  BigInt.asUintN ( bits, bigint )
+
+  1. Let bits be ? ToIndex(bits).
+  2. Let bigint ? ToBigInt(bigint).
+
+features: [BigInt]
+---*/
+
+var i = 0;
+var bits = {
+  valueOf() {
+    assert.sameValue(i, 0);
+    i++;
+    return 0;
+  }
+};
+var bigint = {
+  valueOf() {
+    assert.sameValue(i, 1);
+    i++;
+    return 0n;
+  }
+};
+
+BigInt.asUintN(bits, bigint);
```

Comment:
For good measure, you could assert that the value here is 0n (right?).

Reply:
Sure. I omitted that assert because it's not what this test is really about. I thought it would be clearer to a reader if they see that all the asserts have to do with execution order. But of course adding more asserts doesn't hurt.

I'd prefer to leave it the way it is, but I can change it if others think that would be better.